### ISSUE TITLE
Release 3.0.3

### ISSUE THREAD

**Issue by headius** [ruby/psych#352]
I need to release 3.0.3 to address jruby/jruby#5098.

I will do so today if there are no objections. cc @tenderlove @hsbt.

**Comment by tenderlove** [comment]
Go for it!

**Comment by headius** [comment]
Given that there's been no other commits in almost a month, I shall for it indeed be going!

**Comment by headius** [comment]
I need to push a pre gem so we can push and iterate fixes on a JRuby branch that updates Psych and its tests. There appears to be some regressions, either due to new tests not fixed in the JRuby parts of Psych, or due to behavioral changes in SnakeYAML. See jruby/jruby#5142.

**Comment by headius** [comment]
Excludes for these failures, should they be useful:
```diff
commit dbd70b4930477f4e6efba07a0a6a05f4765334bb
Author: Charles Oliver Nutter <headius@headius.com>
Date:   Mon Apr 16 11:47:36 2018 -0500

    Tag off new Psych 3 failures.

diff --git a/test/mri/excludes/Psych/JSON/TestStream.rb b/test/mri/excludes/Psych/JSON/TestStream.rb
new file mode 100644
index 0000000000..a433613abc
--- /dev/null
+++ b/test/mri/excludes/Psych/JSON/TestStream.rb
@@ -0,0 +1,4 @@
+exclude :test_datetime, "new test or regression, #5142"
+exclude :test_hash, "new test or regression, #5142"
+exclude :test_list_to_json, "new test or regression, #5142"
+exclude :test_time, "new test or regression, #5142"
diff --git a/test/mri/excludes/Psych/TestCoder.rb b/test/mri/excludes/Psych/TestCoder.rb
new file mode 100644
index 0000000000..54282f5aac
--- /dev/null
+++ b/test/mri/excludes/Psych/TestCoder.rb
@@ -0,0 +1 @@
+exclude :test_dump_with_tag, "new test or regression, #5142"
diff --git a/test/mri/excludes/Psych/TestJSONTree.rb b/test/mri/excludes/Psych/TestJSONTree.rb
new file mode 100644
index 0000000000..a433613abc
--- /dev/null
+++ b/test/mri/excludes/Psych/TestJSONTree.rb
@@ -0,0 +1,4 @@
+exclude :test_datetime, "new test or regression, #5142"
+exclude :test_hash, "new test or regression, #5142"
+exclude :test_list_to_json, "new test or regression, #5142"
+exclude :test_time, "new test or regression, #5142"
diff --git a/test/mri/excludes/Psych/TestOmap.rb b/test/mri/excludes/Psych/TestOmap.rb
new file mode 100644
index 0000000000..d5e2564a8f
--- /dev/null
+++ b/test/mri/excludes/Psych/TestOmap.rb
@@ -0,0 +1 @@
+exclude :test_dump, "new test or regression, #5142"
diff --git a/test/mri/excludes/Psych/TestToYamlProperties.rb b/test/mri/excludes/Psych/TestToYamlProperties.rb
new file mode 100644
index 0000000000..09e5fd915e
--- /dev/null
+++ b/test/mri/excludes/Psych/TestToYamlProperties.rb
@@ -0,0 +1,3 @@
+exclude :test_object_dump_yaml_properties, "new test or regression, #5142"
+exclude :test_string_dump, "new test or regression, #5142"
+exclude :test_struct_dump_yaml_properties, "new test or regression, #5142"
diff --git a/test/mri/excludes/Psych/Visitors/TestEmitter.rb b/test/mri/excludes/Psych/Visitors/TestEmitter.rb
new file mode 100644
index 0000000000..16e63c0256
--- /dev/null
+++ b/test/mri/excludes/Psych/Visitors/TestEmitter.rb
@@ -0,0 +1,2 @@
+exclude :test_options, "new test or regression, #5142"
+exclude :test_sequence, "new test or regression, #5142"
```

**Comment by hsbt** [comment]
@headius Do release 3.0.3 :octocat:

**Comment by kares** [comment]
@headius please include https://github.com/ruby/psych/pull/365 in 3.0.3

**Comment by headius** [comment]
@kares I have merged in #365 and we just need @hsbt to agree to a release of psych 3.1 (or we need to merge the relevant changes back to a 3.0 branch.

@hsbt We would like to resolve this for JRuby 9.2.1, but we need a newer release than psych 3.0.2 to do it. Can that happen soon?

**Comment by hsbt** [comment]
@headius The current master branch targets 3.1 release. I will create 3.0.3 tag picked #365 and release it. Please wait a few days.

**Comment by hsbt** [comment]
@headius I just released 3.0.3.pre2

* https://rubygems.org/gems/psych/versions/3.0.3.pre2-java
* https://github.com/ruby/psych/releases/tag/v3.0.3.pre2

Can you confirm them? After that, I will release 3.0.3.

**Comment by headius** [comment]
@headius I will confirm today!

**Comment by headius** [comment]
@hsbt I fixed an issue in 2261525efa9e32ea2556ad3bc34ead71cada5315 and confirmed it works properly in JRuby. Can you re-spin a preview and I'll check that it installs and runs correctly before final release? I promise to be quick about it :-)

**Comment by hsbt** [comment]
@headius Thanks. I pushed 3.0.3.pre3 includes https://github.com/ruby/psych/commit/2261525efa9e32ea2556ad3bc34ead71cada5315.

**Comment by hsbt** [comment]
@headius Just released 3.0.3. Please use it. 

https://rubygems.org/gems/psych/versions/3.0.3-java
